server/src/infra/ldap_handler.rs

```diff
@@ -413,16 +418,33 @@ impl<Backend: BackendHandler + LoginHandler + OpaqueHandler> LdapHandler<Backend
 
     #[instrument(skip_all, level = "debug")]
     async fn do_extended_request(&mut self, request: &LdapExtendedRequest) -> Vec<LdapOp> {
-        match LdapPasswordModifyRequest::try_from(request) {
-            Ok(password_request) => self
+        if let Ok(password_request) = LdapPasswordModifyRequest::try_from(request) {
```

Comment:
I think I understand better why there's no try_from for the who am I request: I think the correct pattern is to match on the name (OID), then try to parse the matching request. I understand if you don't want to do this refactoring (you already did more than I expected!) but I think the correct pattern to follow is:

```rust
match (request.name) {
  OID_PASSWORD_MODIFY => try to parse the request, and error otherwise,
  OID_WHOAMI => parsing is easy :D
  _ => unwilling to perform
}
```